stored_requests/backends/file_fetcher/fetcher_test.go

```diff
@@ -97,63 +104,64 @@ func assertErrorCount(t *testing.T, num int, errs []error) {
 	}
 }
 
-func newCategoryFetcher(directory string) (stored_requests.CategoryFetcher, error) {
+func newCategoryFetcher(t *testing.T, directory string) stored_requests.CategoryFetcher {
 	fetcher, err := NewFileFetcher(directory)
 	if err != nil {
-		return nil, err
+		t.Errorf("Failed to create a category Fetcher: %v", err)
 	}
 	catfetcher, ok := fetcher.(stored_requests.CategoryFetcher)
 	if !ok {
-		return nil, fmt.Errorf("Failed to type cast fetcher to CategoryFetcher")
+		t.Errorf("Failed to type cast fetcher to CategoryFetcher")
 	}
-	return catfetcher, nil
+	return catfetcher
+}
+
+func TestCategoriesFetcherAllCategories(t *testing.T) {
+	fetcher := newCategoryFetcher(t, "./test/category-mapping")
+	categories, errs := fetcher.FetchAllCategories(nil)
+	assertErrorCount(t, 1, errs)
+	assert.EqualError(t, errs[0], `Unable to unmarshal categories from "test/category-mapping/test/test_broken.json"`)
+	assert.Equalf(t, len(categories), 2, "Expected 2 categories preloaded, got %d", len(categories))
+	assert.Containsf(t, categories, "test/test", "test/test missing from preloaded data set")
+	assert.Containsf(t, categories, "test/test_categories", "test/test_categories missing from preloaded data set")
+	cat, err := fetcher.FetchCategories(nil, "test", "categories", "IAB1-1")
```

Comment:
Any reason for incorporating the `FetchCategories` test as well here? These should be covered in the `FetchCategories` specific code down below, no?

Reply:
It was intentional, it is the only way to check that the preloaded categories were actually decoded and work as expected.